app/services/post_alerter.rb

```diff
@@ -553,7 +603,10 @@ def extract_mentions(post)
       users = nil if users.empty?
     end
 
-    [groups, users]
+    # @here can be a user mention and then this feature is disabled
+    here = mentions.include?(SiteSetting.here_mention) && Guardian.new(post.user).can_mention_here?
```

Comment:
I think we should update the username validator to disallow usernames as part of `SiteSetting.here_mention`.

Reply:
I did that here where we check the other reserved usernames:

https://github.com/discourse/discourse/pull/14900/files#diff-9802ca3c9c4cf89904fd44bc114e35ebdf2c5dd3d5b645491e2b253e1afef29bR288-R289

Reply:
Ah I must have missed it while reviewing. Resolving this :)